compiler/rustc_feature/src/unstable.rs

```diff
@@ -36,35 +36,64 @@ macro_rules! status_to_enum {
 #[derive(Clone, Default, Debug)]
 pub struct Features {
     /// `#![feature]` attrs for language features, for error reporting.
-    enabled_lang_features: Vec<(Symbol, Span, Option<Symbol>)>,
+    enabled_lang_features: Vec<EnabledLangFeature>,
     /// `#![feature]` attrs for non-language (library) features.
-    enabled_lib_features: Vec<(Symbol, Span)>,
+    enabled_lib_features: Vec<EnabledLibFeature>,
     /// `enabled_lang_features` + `enabled_lib_features`.
     enabled_features: FxHashSet<Symbol>,
 }
 
+/// Information about an enabled language feature.
+#[derive(Debug, Copy, Clone)]
+pub struct EnabledLangFeature {
+    /// Name of the feature gate guarding the language feature.
+    pub gate_name: Symbol,
+    /// Span of the `#[feature(...)]` attribute.
+    pub attr_sp: Span,
+    /// Stability info of the language feature.
+    pub stability: LangFeatureStability,
```

Comment:
Note that this is only used for the lint that shows "this feature is already stable" when a `feature` attribute is found for a stable lang feature. Not sure if that's worth its own dedicated type...

Reply:
Yeah, I can drop this as a specific type, it seemed a bit overkill when I added it

Reply:
Dropped this because it seemed like unnecessary complexity.